### ISSUE TITLE
Videos are uploaded successfully, but cannot be accessed on the dashboard

### ISSUE THREAD

**Issue by muhammad-st** [cypress-io/cypress#17833]
### Current behavior

After a failing test, In the log, it says "uploaded successfully":
```
(Uploading Results)

- Done Uploading (1/1) test file name here -- test case name here (failed).png
```
Meanwhile, if you try to open the video on the dashboard, it won't open, displays the following:
```
The media could not be loaded, either because the server or network failed or because the format is not supported.
```
Internally, the video file path returns:
```
<Error>
<script/>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>screencasts/global/905651e6-b11e-4fa9-9ce8-68e76f4a2dfb.mp4</Key>
<RequestId>....</RequestId>
<HostId>....</HostId>
</Error>
```

### Desired behavior

The Video should be accessible and working.

### Test code to reproduce

- connect the project to cyrpess.io
- run multiple tests containing at least one failing/flaky test
- check the video on the dashboard of cypress.io

### Cypress Version

8.3.0

### Other

it happens on `8.2.0` also.

**Comment by jennifer-shehane** [comment]
The log you posted is for the screenshot uploading. There should be a separate log below that indicating that it’s compressing and uploading the video. Is there no log displaying that?

Do you have any configuration settings around video like the video key or compression or videoUpload...?

**Comment by muhammad-st** [comment]
Ops sorry didn't notice the `.png` extension.

Searched the log for any `compress` occurrence, not found.

The process exited normally with:
```
[2021-08-19T20:49:41.886Z]   (Run Finished)
[2021-08-19T20:49:41.886Z] 
[2021-08-19T20:49:41.886Z] 
                                            TABLE HERE
[2021-08-19T20:49:41.886Z] 
[2021-08-19T20:49:41.886Z] 
[2021-08-19T20:49:41.886Z] ───────────────────────────────────────────────────────────────────────────────────────────────────────
[2021-08-19T20:49:41.886Z]                                                                                                        
[2021-08-19T20:49:41.886Z]   Recorded Run: run url...                             
[2021-08-19T20:49:41.886Z] 
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:41.886Z] Successfully deleted passing test file_name....
[2021-08-19T20:49:42.142Z] Done in 1241.23s.
```

cypress.json
```json
{
    "viewportWidth": 1920,
    "viewportHeight": 1080,
    "videoUploadOnPasses": false,
    "baseUrl": "url...",
    "defaultCommandTimeout": 20000,
    "projectId": "id...",
    "retries": {
      "runMode": 1,
      "openMode": 0
    }
  }
  
```

**Comment by jennifer-shehane** [comment]
At the end of the test, there should be a log with the table like shown below. Can you include what is printed in this area please when the video is not available? Or include the entire log. 

```

  20 passing (1m)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        20                                                                               │
  │ Passing:      20                                                                               │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     1 minute, 5 seconds                                                              │
  │ Estimated:    1 minute, 2 seconds                                                              │
  │ Spec Ran:     ui/transaction-feeds.spec.ts                                                     │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Video)

  -  Started processing:  Compressing to 32 CRF
    Compression progress:  56%
  -  Finished processing: /home/person/cypress-realworld-app/cypress/videos/ui/transa   (19 seconds)
                          ction-feeds.spec.ts.mp4


  (Uploading Results)

  - Done Uploading (1/1) /home/person/cypress-realworld-app/cypress/videos/ui/transaction-feeds.spec.ts.mp4

====================================================================================================
```

**Comment by muhammad-st** [comment]
Yea, the whole section of `(Video)` was not present.

> Do you have any configuration settings around video like the video key or compression or videoUpload...?

Thanks to your hint (above), I discovered that Cypress surprisingly doesn't upload flaky tests, and you have some documentation in this regard created along with the issue I linked in my previous message/post.

For me, the issue has been resolved by adding this snippet. Still, the issue that's not resolved maybe for other people whining to use `videoUploadOnPasses: false` is this #2939 which happened to me in the case of "flaky tests".
